samples/bluetooth/direct_test_mode/src/dtm.c

```diff
@@ -889,10 +942,57 @@ int dtm_init(void)
 
 	dtm_inst.state = STATE_IDLE;
 	dtm_inst.packet_len = 0;
+	dtm_inst.cte_info.iq_rep_cb = callback;
 
 	return 0;
 }
 
+#if DIRECTION_FINDING_SUPPORTED
+static void report_iq(void)
+{
+	struct dtm_iq_data iq_data;
+
+	iq_data.channel = dtm_inst.phys_ch;
+	iq_data.rssi = -nrf_radio_rssi_sample_get(NRF_RADIO);
+	nrf_radio_event_clear(NRF_RADIO, NRF_RADIO_EVENT_RSSIEND);
+	iq_data.rssi_ant = dtm_hw_radio_pdu_antenna_get();
+
+	if (dtm_inst.cte_info.mode == DTM_CTE_MODE_AOD) {
+		if (dtm_inst.cte_info.slot == DTM_CTE_SLOT_1US) {
+			iq_data.type = DTM_CTE_TYPE_AOD_1US;
+		} else if (dtm_inst.cte_info.slot == DTM_CTE_SLOT_2US) {
+			iq_data.type = DTM_CTE_TYPE_AOD_2US;
+		} else {
+			/* Not possible - invalid value */
+			__ASSERT_NO_MSG(false);
+		}
+	} else if (dtm_inst.cte_info.mode == DTM_CTE_MODE_AOA) {
+		iq_data.type = DTM_CTE_TYPE_AOA;
+	} else {
+		/* Not possible - invalid value */
+		__ASSERT_NO_MSG(false);
+	}
+
+	if (dtm_inst.cte_info.slot == DTM_CTE_SLOT_1US) {
+		iq_data.slot = DTM_CTE_SLOT_DURATION_1US;
+	} else if (dtm_inst.cte_info.slot == DTM_CTE_SLOT_2US) {
+		iq_data.slot = DTM_CTE_SLOT_DURATION_2US;
+	} else {
+		/* Not possible - invalid value */
+		__ASSERT_NO_MSG(false);
+	}
+
+	/* There is no requirement to report iq samples with invalid CRC */
+	iq_data.status = DTM_PACKET_STATUS_CRC_OK;
```

Comment:
How do you know that status is ok?

Reply:
The IQ sampling does not need to report incorrect CRC samples. In the dtm.c file the reporting function is called only if CRC is correct.